builders/build/moonbeam-custom-api.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Moonbeam Custom API
+description: This page contains information on Moonbeam custom API endpoints. 
+---
+
+# Moonbeam Custom API 
+
+![Moonbeam Custom API](/images/builders/build/moonbeam-custom-api/moonbeam-custom-api-banner.png)
+
+## Finality RPC Endpoints {: #finality-rpc-endpoints }
+
+Moonbeam node has added support for two custom JSON RPC endpoints, `moon_isBlockFinalized` and `moon_isTxFinalized`, that are useful for checking whether an on-chain event is finalized. The information on these two endpoints are as follows:
+
+=== "moon_isBlockFinalized"
+    |   Variable   |                                      Value                                       |
+    |:------------:|:--------------------------------------------------------------------------------|
+    |   Endpoint |                        `moon_isBlockFinalized`                     |
+    |   Description   | Check for the finality of the block given by its block hash |
+    |  Parameters |    `block_hash`: **STRING** The hash of the block, accepts either Substrate-style or Ethereum-style block hash as its input                     | 
+    |  Returns | `result`: **BOOLEAN** Returns `true` if the block is finalized, `false` if the block is not finalized or not found  | 
+
+=== "moon_isTxFinalized"
+    |   Variable   |                                      Value                                       |
+    |:------------:|:--------------------------------------------------------------------------------|
+    |   Endpoint |                        `moon_isTxFinalized`                     |
+    |   Description   | Check for the finality of the transaction given by its EVM tx hash |
+    |  Parameters | `tx_hash`: **STRING** The EVM tx hash of the transaction  | 
+    |  Returns |  `result`: **BOOLEAN** Returns `true` if the tx is finalized; `false` if the tx is not finalized or not found | 
+
+You can try out these endpoints with the following curl examples. These examples query the public RPC endpoint of Moonbase Alpha, but they can be modified to use with Moonbeam and Moonriver by changing the URL of the RPC endpoint to the corresponding [endpoints](https://docs.moonbeam.network/builders/get-started/endpoints/){target=blank}. 
```

Comment:
I would reword this section a little bit, because they will need to sign up for their own API endpoint/api key... maybe something like...

`... but they can be modified to use with Moonbeam and Moonriver by changing the URL of the RPC endpoint to use your own endpoint and API key which you can get from one of the supported [Endpoint Providers](/builders/get-started/endpoints/){target=_blank}`